Client.java

```diff
@@ -1,80 +1,56 @@
 import java.util.*;
-import java.io.*;
 
 public class Client{
-   public static void main(String[] args){
+
+   private static final String[] PLAYER_NAMES = {"Bob", "Joe", "Bill", "Kevin"};
+
+//   private static final int MAX_MOVES = 1000;
+
+   private static void nextMoveFor(Player player)
+   {
+      int spin = Board.spin();
+      System.out.println(player.getName() + " spins: " + spin);
+      player.setSpace(Board.getNewLocation(player.getSpace(), spin));
+      System.out.println(player + "\n\n");
+   }
+   public static void main(String[] args)
+   {
       Scanner input = new Scanner(System.in);
       System.out.println("How many players (2 - 4)?");
-      int plrNum = input.nextInt();
-      
-      Player p1 = new Player("Bob");
-      Player p2 = new Player("Joe");
-      Player p3 = new Player("Bill");
-      Player p4 = new Player("Kevin");
-      
-      Board b = new Board();
-      
-      while(p1.getSpace() < 100 && p2.getSpace() < 100 && p3.getSpace() < 100 && p4.getSpace() < 100){
-      
-         if(plrNum >= 1){
-            int spin = b.spin();
-            System.out.println(p1.getName() + " spins: " + spin);
-            int newLoc = b.getNewLocation(p1.getSpace(), spin);
-            p1.setSpace(newLoc);
-            System.out.println(p1.getName() + " is on space " + p1.getSpace());  
-            System.out.println("");
-            System.out.println("");
-         }
-         
-         if(p1.getSpace() == 100){
-            System.out.println(p1.getName() + " wins!");
-            break;
-         }
-         
-         if(plrNum >= 2){
-            int spin = b.spin();
-            System.out.println(p2.getName() + " spins: " + spin);
-            int newLoc = b.getNewLocation(p2.getSpace(), spin);
-            p2.setSpace(newLoc);
-            System.out.println(p2.getName() + " is on space " + p2.getSpace());  
-            System.out.println("");
-            System.out.println("");
-         }
-         
-         if(p2.getSpace() == 100){
-            System.out.println(p2.getName() + " wins!");
-            break;
-         }
-         
-         if(plrNum >= 3){
-            int spin = b.spin();
-            System.out.println(p3.getName() + " spins: " + spin);
-            int newLoc = b.getNewLocation(p3.getSpace(), spin);
-            p3.setSpace(newLoc);
-            System.out.println(p3.getName() + " is on space " + p3.getSpace());  
-            System.out.println("");
-            System.out.println("");
-         }
-         
-         if(p3.getSpace() == 100){
-            System.out.println(p3.getName() + " wins!");
-            break;

```

Comment:
u c... originally u were doing a comparison here 2 check if a player won...
but the `break` was kind of pointless since u would b doing effectively the same check again in the `while` loop condition.